File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorDateFkDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_date_fk. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorDateFkDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorDateFkDbm _instance = new VendorDateFkDbm();
    private VendorDateFkDbm() {}
    public static VendorDateFkDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorDateFk)et).getBarId(), (et, vl) -> ((VendorDateFk)et).setBarId(cti(vl)), "barId");
        setupEpg(_epgMap, et -> ((VendorDateFk)et).getBarDate(), (et, vl) -> ((VendorDateFk)et).setBarDate(ctld(vl)), "barDate");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((VendorDateFk)et).getVendorDatePk(), (et, vl) -> ((VendorDateFk)et).setVendorDatePk((OptionalEntity<VendorDatePk>)vl), "vendorDatePk");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_date_fk";
    protected final String _tableDispName = "vendor_date_fk";
    protected final String _tablePropertyName = "vendorDateFk";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_date_fk", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnBarId = cci("bar_id", "bar_id", null, null, Integer.class, "barId", null, true, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnBarDate = cci("bar_date", "bar_date", null, null, java.time.LocalDate.class, "barDate", null, false, false, true, "date", 13, 0, null, false, null, null, "vendorDatePk", null, null, false);
//"df.Date"

    /**
     * bar_id: {PK, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnBarId() { return _columnBarId; }
    /**
     * bar_date: {NotNull, date(13), FK to vendor_date_pk}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnBarDate() { return _columnBarDate; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnBarId());
        ls.add(columnBarDate());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnBarId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * vendor_date_pk by my bar_date, named 'vendorDatePk'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorDatePk() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnBarDate(), VendorDatePkDbm.getInstance().columnFooDate());
        return cfi("fk_vendor_date_fk_pk", "vendorDatePk", this, VendorDatePkDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorDateFkList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorDateFk"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorDateFkCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorDateFkBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorDateFk> getEntityType() { return VendorDateFk.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorDateFk newEntity() { return new VendorDateFk(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorDateFk)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorDateFk)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/vendorDateFkdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorDateFkDbm_T struct {
	df.BaseDBMeta
	ColumnBarId *df.ColumnInfo
	ColumnBarDate *df.ColumnInfo
}

func (b *VendorDateFkDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorDateFkDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorDateFkDbm *VendorDateFkDbm_T

func Create_VendorDateFkDbm() {
	VendorDateFkDbm = new(VendorDateFkDbm_T)
	VendorDateFkDbm.TableDbName = "vendor_date_fk"
	VendorDateFkDbm.TableDispName = "vendor_date_fk"
	VendorDateFkDbm.TablePropertyName = "vendorDateFk"
	VendorDateFkDbm.TableSqlName = new(df.TableSqlName)
	VendorDateFkDbm.TableSqlName.TableSqlName = "vendor_date_fk"
	VendorDateFkDbm.TableSqlName.CorrespondingDbName = VendorDateFkDbm.TableDbName

	var vendorDateFk df.DBMeta
	vendorDateFk = VendorDateFkDbm
	VendorDateFkDbm.DBMeta=&vendorDateFk
	barIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo bar_id
	barIdSqlName.ColumnSqlName = "bar_id"
	barIdSqlName.IrregularChar = false
	VendorDateFkDbm.ColumnBarId = df.CCI(&vendorDateFk, "bar_id", barIdSqlName, "", "", "Integer.class", "barId", "", true, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	barDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo bar_date
	barDateSqlName.ColumnSqlName = "bar_date"
	barDateSqlName.IrregularChar = false
	VendorDateFkDbm.ColumnBarDate = df.CCI(&vendorDateFk, "bar_date", barDateSqlName, "", "", "java.time.LocalDate.class", "barDate", "", false, false,true, "date", 13, 0, "",false,"","", "vendorDatePk","","",false,"df.Date")

	VendorDateFkDbm.ColumnInfoList = new(df.List)
	VendorDateFkDbm.ColumnInfoList.Add(VendorDateFkDbm.ColumnBarId)
	VendorDateFkDbm.ColumnInfoList.Add(VendorDateFkDbm.ColumnBarDate)


	VendorDateFkDbm.ColumnInfoMap=make(map[string]int)
	VendorDateFkDbm.ColumnInfoMap["barId"]=0
		VendorDateFkDbm.ColumnInfoMap["barDate"]=1
	    VendorDateFkDbm.PrimaryKey = true
    VendorDateFkDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorDateFk
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorDateFkDbm.ColumnBarId)

	VendorDateFkDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorDateFkDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorDateFkMeta df.DBMeta = VendorDateFkDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorDateFk"] = &vendorDateFkMeta
}
